plugins/jobtype/src/azkaban/jobtype/tuning/TuningParameterUtils.java

```diff
@@ -0,0 +1,232 @@
+  private static final String AZKABAN = "azkaban";
+  private static final String TUNING_API_VERSION = "2";
+
+  private TuningParameterUtils() {
+
+  }
+
+  /**
+   * Get hadoop properties from the job
+   * @param props
+   * @return Map of parameter name and value
+   */
+  public static Map<String, String> getHadoopProperties(Props props) {
+    Map<String, String> confProperties = props.getMapByPrefix(HadoopConfigurationInjector.INJECT_PREFIX);
+    if (log.isDebugEnabled()) {
+      for (Map.Entry<String, String> entry : confProperties.entrySet()) {
+        log.debug("Adding default parameter " + entry.getKey() + ", value " + entry.getValue());
+      }
+    }
+    return confProperties;
+  }
+
+  /**
+   * This method calls Auto Tuning Framework Rest API and adds new parameters in props.
+   * This method will retry 3 times in case there is timeout while calling rest API.
+   * @param props
+   */
+  public static void updateAutoTuningParameters(Props props) {
+    int retryCount = 0;
+    boolean noException = false;
+    Map<String, String> params = null;
+    if (!props.containsKey(TuningCommonConstants.TUNING_API_END_POINT)) {
+      log.error("Tuning API End Point not found in properties. Not applying auto tuning. ");
+      return;
+    }
+    int maxRetryCount = DEFAULT_TUNING_API_RETRY_COUNT;
+    if (props.containsKey(TUNING_API_RETRY_COUNT)) {
+      maxRetryCount = props.getInt(TUNING_API_RETRY_COUNT);
+    }
+
+    while (!noException && retryCount < maxRetryCount) {
+      try {
+        retryCount++;
+        log.info("Calling get current run parameters. Try count " + retryCount);
+        params = getCurrentRunParameters(props);
+        noException = true;
+      } catch (Exception e) {
+        log.error("Error in getting current run parameter ", e);
+      }
+    }
+    //In case parameters are null, do nothing and job will run with default parameters of the job
+    if (params != null) {
+      for (Map.Entry<String, String> param : params.entrySet()) {
+        props.put(HadoopConfigurationInjector.INJECT_PREFIX + param.getKey(), param.getValue());
+        log.info("Applied parameter " + param.getKey() + " with value " + param.getValue());
+      }
+      props.put(HadoopConfigurationInjector.INJECT_PREFIX + TuningCommonConstants.AUTO_TUNING_ENABLED, "true");
+    }
+  }
+
+  /**
+   *This method is for getting current run parameters from Dr Elephant
+   * @param props
+   * @return Map of parameter name and value
+   * @throws IOException
+   * @throws ParseException
+   */
+  @SuppressWarnings({ "unchecked" })
+  public static Map<String, String> getCurrentRunParameters(Props props) throws IOException {
+
+    log.debug("Properties are " + props.toString());
+
+    CloseableHttpClient httpClient = HttpClients.createDefault();
+
+    HttpResponse response = null;
+    String endPoint = String.format("%s", props.get(TuningCommonConstants.TUNING_API_END_POINT));
```

Comment:
What is being achieved with this line?

Reply:
Earlier was using multiple string concatenation. Removed the string.format. Directly using property.